common/fallible/src/copy_from_slice.rs

```diff
@@ -0,0 +1,17 @@
+// Copyright (c) The Diem Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+const LENGTH_MISMATCH: &str = "can't copy source slice into destination slice: sizes don't match";
+
```

Comment:
You should make this a type and impl Error on it (using that string for its Display impl)

Reply:
The error should also include the lengths of the slice and the vec.

Reply:
used thiserror to do that : )